### PR TITLE
Fixes #199 Fix failing tests for Scala 2.11

### LINKED ISSUE #199: Scala: Test failures in adts
There are test failures in the scala section in atds (however possibly specific to the version of Scala used).

To reproduce (on `ocaml/opam2` docker image):
`docker run -it ocaml/opam2:ubuntu bash`
Install dependencies:
```bash
opam install dune
cd ..
git clone https://github.com/ahrefs/atd.git
cd atd/
dune external-lib-deps --missing @@default
opam install biniou easy-format re yojson
sudo apt-get install m4
opam install menhir
sudo apt-get install scala
sudo apt-get install openjdk-11-jdk
```

Build and run tests;
```bash
opam@216b8d8251bd:~/atd$ make clean                                 
dune clean
opam@216b8d8251bd:~/atd$ make all
dune build
opam@216b8d8251bd:~/atd$ make test
dune runtest
test_atdgen_main alias atdgen/test/runtest
Warning: skipping field b2 (def: File "test.atd", line 98, characters 12-40)
Warning: skipping field b3 (def: File "test.atd", line 98, characters 12-40)
Warning: skipping field b4 (def: File "test.atd", line 98, characters 12-40)
Warning: skipping field b5 (def: File "test.atd", line 98, characters 12-40)
    run_test alias atds/test/runtest (exit 1)
(cd _build/default/atds/test && ./run_test.sh)
test.scala:64: error: identifier expected but ')' found.
) extends Atds {
^
test.scala:241: error: ':' expected but '}' found.
}
^
two errors found
make: *** [Makefile:7: test] Error 1
```

Cause for this was a comma in some inappropriate places, like this:
```scala
...
  final_ : Option[Int],
  kase : String,
  l2 : List[RecordWithDefaults],
) extends Atds {
...
```

After fixing `test.scala` in the `_build` directory, the file compiled with these changes:
```patch
opam@216b8d8251bd:~/atd/_build/default/atds/test$ diff test.expected.scala test.scala 
63c63
<   opt : Option[Boolean],
---
>   opt : Option[Boolean]
68c68
<     "opt" := opt,
---
>     "opt" := opt
141c141
<   l2 : List[RecordWithDefaults],
---
>   l2 : List[RecordWithDefaults]
154c154
<     "l2" := l2,
---
>     "l2" := l2
164c164
<   e : E,
---
>   e : E
173c173
<     "e" := e,
---
>     "e" := e
184c184
<   more : String,
---
>   more : String
194c194
<     "more" := more,
---
>     "more" := more
231c231
<   state : String,
---
>   state : String
237c237
<     "state" := state,
---
>     "state" := state
```
(Note that the misplaced commas are also present in `test.expected.scala`)

However, the scala tests still failed:
```
opam@216b8d8251bd:~/atd/_build/default/atds/test$ ./run_test.sh                       
JUnit version 4.8.2
.E.E.E
Time: 0.101
There were 3 failures:
1) testDefaults(AtdsTest)
java.lang.NoSuchMethodError: 'void scala.Product.$init$(scala.Product)'
        at argonaut.Json.<init>(Json.scala:13)
        at argonaut.JNull$.<init>(Json.scala:479)
        at argonaut.JNull$.<clinit>(Json.scala)
        ...
```

...which however might be specific to versions of dependencies and/or versions of Scala and Java used here:
```bash
opam@216b8d8251bd:~/atd/_build/default/atds/test$ scala
Welcome to Scala 2.11.12 (OpenJDK 64-Bit Server VM, Java 11.0.4).
```
and `openjdk-11-jdk`, as stated above.

I will look into dependency versions to see how to fix the latter issue.

### PR DESCRIPTION
**Pull request by syntakker**
Commit ba90293, "Fix extra commas for scala":
Fix trailing commas in generated Scala code

Commit c789b2e, "Fix atds tests for Scala 2.11":
- argonaut_2.11-6.2.2.jar added in `atds/test/`
- argonaut_2.11-6.2.2.jar added in `deps` in `atds/test/dune`
- `CLASSPATH` in `atds/test/run_test.sh` set according to installed Scala version

This fixes the tests in `atds` for me.
Tested with Scala versions 2.11.12 and 2.12.3

Fixes https://github.com/ahrefs/atd/issues/199

### CODE REVIEW

File: atd/src/annot.ml [hunk 1]
```diff
@@ -10,8 +10,20 @@ let error_at loc s =
...
+      List.filter_map (fun (f, (l, s)) ->
+        if f = field then Some (l, s) else None)
+      fs)
+    |> List.flatten in
+  match fieldmatches with
+  | [fieldmatch] -> Some fieldmatch
+  | (loc, _) :: others -> error_at loc
+    (sprintf "Duplicate annotation %s.%s (also in:\n  %s\n)" section field
+    (List.map (fun (loc, _) -> (Ast.string_of_loc loc)) others
+     |> String.concat ",/n  "))
```
Comment (rgrinberg) [inline atd/src/annot.ml]: This `/n` looks a bit weird. Shouldn't it be a backslash?

Comment (syntakker) [reply atd/src/annot.ml]: Absolutely. I'll fix it and complete the fix by tests.

I neglected that pull requests are branch-wise. I originally wanted to have the commits for https://github.com/ahrefs/atd/issues/89 in an extra pull request. This commit slipped in somewhat accidentally. I better separate the two issues.

Comment (syntakker) [reply atd/src/annot.ml]: Moved the fixes for https://github.com/ahrefs/atd/issues/89 to a branch of its own, see https://github.com/ahrefs/atd/pull/201.
Backslash is fixed, tests are added.

**Comment by rgrinberg** [comment]
I merged this manually and excluded the reverted commits. Thanks

**Pull request closed without merging**